homeassistant/components/ads/cover.py

```diff
@@ -0,0 +1,175 @@
+    vol.Optional(CONF_ADS_VAR): cv.string,
+    vol.Optional(CONF_ADS_VAR_POSITION): cv.string,
+    vol.Optional(CONF_ADS_VAR_SET_POS): cv.string,
+    vol.Optional(CONF_ADS_VAR_CLOSE): cv.string,
+    vol.Optional(CONF_ADS_VAR_OPEN): cv.string,
+    vol.Optional(CONF_ADS_VAR_STOP): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the cover platform for ADS."""
+    ads_hub = hass.data.get(DATA_ADS)
+
+    ads_var_is_closed = config.get(CONF_ADS_VAR)
+    ads_var_position = config.get(CONF_ADS_VAR_POSITION)
+    ads_var_pos_set = config.get(CONF_ADS_VAR_SET_POS)
+    ads_var_open = config.get(CONF_ADS_VAR_OPEN)
+    ads_var_close = config.get(CONF_ADS_VAR_CLOSE)
+    ads_var_stop = config.get(CONF_ADS_VAR_STOP)
+    name = config.get(CONF_NAME)
+    device_class = config.get(CONF_DEVICE_CLASS)
+
+    add_entities([AdsCover(ads_hub,
+                           ads_var_is_closed,
+                           ads_var_position,
+                           ads_var_pos_set,
+                           ads_var_open,
+                           ads_var_close,
+                           ads_var_stop,
+                           name,
+                           device_class)])
+
+
+class AdsCover(AdsEntity, CoverDevice):
+    """Representation of ADS cover."""
+
+    def __init__(self, ads_hub,
+                 ads_var_is_closed, ads_var_position,
+                 ads_var_pos_set, ads_var_open,
+                 ads_var_close, ads_var_stop, name, device_class):
+        """Initialize AdsCover entity."""
+        super().__init__(ads_hub, name, ads_var_is_closed)
+        if self._ads_var is None:
+            if ads_var_position is not None:
+                self._unique_id = ads_var_position
+            elif ads_var_pos_set is not None:
+                self._unique_id = ads_var_pos_set
+            elif ads_var_open is not None:
+                self._unique_id = ads_var_open
+
+        self._state_dict[STATE_KEY_POSITION] = None
+        self._ads_var_position = ads_var_position
+        self._ads_var_pos_set = ads_var_pos_set
+        self._ads_var_open = ads_var_open
+        self._ads_var_close = ads_var_close
+        self._ads_var_stop = ads_var_stop
+        self._device_class = device_class
+
+    async def async_added_to_hass(self):
+        """Register device notification."""
+        if self._ads_var is not None:
+            await self.async_initialize_device(self._ads_var,
+                                               self._ads_hub.PLCTYPE_BOOL)
+
+        if self._ads_var_position is not None:
+            await self.async_initialize_device(self._ads_var_position,
+                                               self._ads_hub.PLCTYPE_BYTE,
+                                               STATE_KEY_POSITION)
+
+    @property
+    def state(self):
```

Comment:
Don't overwrite this. `state` property is handled by the base cover entity class.